packages/editor-ui/src/stores/logStreaming.store.ts

```diff
@@ -32,221 +33,244 @@ export interface DestinationSettingsStore {
+				items[destination.id]?.selectedEvents,
+			);
+		}
+	};
+
+	const getDestination = (destinationId: string) => {
+		if (items[destinationId]) {
+			return items[destinationId].destination;
+		} else {
+			return;
+		}
+	};
+
+	const getAllDestinations = () => {
+		const destinations: MessageEventBusDestinationOptions[] = [];
+		for (const key of Object.keys(items)) {
+			destinations.push(items[key].destination);
+		}
+		return destinations;
+	};
+
+	const clearDestinations = () => {
+		Object.keys(items).forEach((key) => delete items[key]);
+	};
+
+	const addEventName = (name: string) => {
+		eventNames.value.add(name);
+	};
+
+	const removeEventName = (name: string) => {
+		eventNames.value.delete(name);
+	};
+
+	const clearEventNames = () => {
+		eventNames.value.clear();
+	};
+
+	const addSelectedEvent = (id: string, name: string) => {
+		items[id]?.selectedEvents?.add(name);
+		setSelectedInGroup(id, name, true);
+	};
+
+	const removeSelectedEvent = (id: string, name: string) => {
+		items[id]?.selectedEvents?.delete(name);
+		setSelectedInGroup(id, name, false);
+	};
+
+	const setSelectedInGroup = (destinationId: string, name: string, isSelected: boolean) => {
+		if (items[destinationId]) {
+			const groupName = eventGroupFromEventName(name);
+			const groupIndex = items[destinationId].eventGroups.findIndex((e) => e.name === groupName);
+
+			if (groupIndex > -1) {
+				if (groupName === name) {
+					items[destinationId].eventGroups[groupIndex].selected = isSelected;
+				} else {
+					const eventIndex = items[destinationId].eventGroups[groupIndex].children.findIndex(
+						(e) => e.name === name,
+					);
+					if (eventIndex > -1) {
+						items[destinationId].eventGroups[groupIndex].children[eventIndex].selected = isSelected;
+						if (isSelected) {
+							items[destinationId].eventGroups[groupIndex].indeterminate = isSelected;
+						} else {
+							let anySelected = false;
+							for (
+								let i = 0;
+								i < items[destinationId].eventGroups[groupIndex].children.length;
+								i++
+							) {
+								anySelected =
+									anySelected || items[destinationId].eventGroups[groupIndex].children[i].selected;
+							}
+							items[destinationId].eventGroups[groupIndex].indeterminate = anySelected;
 						}
 					}
 				}
 			}
-		},
-		removeDestinationItemTree(id: string) {
-			delete this.items[id];
-		},
-		clearDestinationItemTrees() {
-			this.items = {} as DestinationSettingsStore;
-		},
-		setSelectionAndBuildItems(destination: MessageEventBusDestinationOptions) {
-			if (destination.id) {
-				if (!this.items[destination.id]) {
-					this.items[destination.id] = {
-						destination,
-						selectedEvents: new Set<string>(),
-						eventGroups: [],
-						isNew: false,
-					} as DestinationStoreItem;
+		}
+	};
+
+	const removeDestinationItemTree = (id: string) => {
+		delete items[id];
+	};
+
+	const clearDestinationItemTrees = () => {
+		Object.keys(items).forEach((key) => delete items[key]);
```

Comment:
Why not `items = {}`?. Also isn't this exactly same as `clearDestinations`? Maybe we can combine the two